api/placer.py

```diff
@@ -350,23 +352,11 @@ def finalize(self):
                 hierarchy.update_container_hierarchy(self.metadata, bid, self.container_type)
 
         if job_ticket is not None:
-            if success:
-                Queue.mutate(job, {
-                    'state': 'complete',
-                    'profile': {
-                        'elapsed': job_ticket['elapsed']
-                    }
-                })
-            else:
-                Queue.mutate(job, {
-                    'state': 'failed',
-                    'profile': {
-                        'elapsed': job_ticket['elapsed']
-                    }
-                })
+            Queue.mutate(job, {'state': 'complete' if success else 'failed',
+                               'profile': {'elapsed': job_ticket['elapsed']}})
+            job = Job.get(job.id_)
```

Comment:
This will raise an `APINotFoundException`, not return `None`. 

It looks like we call it above too so I'm curious why you're checking for `None` below. Race condition that it gets deleted between the `Queue.mutate` call and `save`?

Reply:
`Queue.mutate` does _not_ update the loaded job's attributes (except partially in this special case: https://github.com/scitran/core/blob/master/api/jobs/queue.py#L82)
Therefore calling `job.save()` once more (when setting `saved_files` below) would try to revert the state transition.
Was thinking of updating `mutate` instead, but wanted to limit the scope of the change and the possible errors introduced.